mitiq/zne/scaling/folding.py

```diff
@@ -296,13 +292,23 @@ def fold_global(
     num_to_fold = int(round(fraction_scale * len(operations) / 2))
 
     if num_to_fold > 0:
-        folded += Circuit(
-            [inverse(operations[-num_to_fold:])], [operations[-num_to_fold:]]
-        )
+        # Create the inverse of the final partial circuit
+        inverse_partial = Circuit()
+        num_partial = 0
+        for moment in base_circuit[::-1]:
+            new_moment = Moment()
+            for op in moment:
+                new_moment = new_moment.with_operation(inverse(op))
+                num_partial += 1
+                if num_partial == num_to_fold:
+                    break
```

Comment:
Must reverse the order of operations in a moment before it gets appended in the new moment. This is what's causing the current test failure. 

Note that the failing test also needs to be corrected to what it was before it was changed for this PR.